src/parser/compile.c

```diff
@@ -746,12 +756,37 @@ pt_add_lock_class (PARSER_CONTEXT * parser, PT_CLASS_LOCKS * lcks, PT_NODE * spe
   synonym_mop = db_find_synonym (class_name);
   if (synonym_mop != NULL)
     {
+      char *r;
+
       class_name = db_get_synonym_target_name (synonym_mop, target_name, DB_MAX_IDENTIFIER_LENGTH);
       if (class_name == NULL)
 	{
 	  ASSERT_ERROR_AND_SET (error);
 	  return error;
 	}
+      if ((r = (char *) strchr (class_name, '@')) != NULL)
+	{
+	  /* remote table */
+	  char *synonym_name, *s;
+
+	  /* for alias */
+	  synonym_name = (char *) spec->info.spec.entity_name->info.name.original;
+	  if ((s = (char *) strchr (synonym_name, '.')) != NULL)
+	    {
+	      synonym_name = s + 1;
+	    }
+
+	  *r = 0;
+	  spec->info.spec.entity_name->info.name.original = pt_append_string (parser, NULL, class_name);
```

Comment:
If target_owner_name is specified in target_unique_name of synonym, owner_name must be skipped in class_name.
```
char *dot = NULL;
if ((dot = (char *) strchr (class_name, '.')) != NULL)
  {
    class_name = dot + 1;
  }
```

Reply:
This logic was reverted by the next commit.